notebooks/classify/signature.py

```diff
@@ -0,0 +1,249 @@
+        
+        for feature in cleanTable:
+            
+            # Check if this feature is included in our exactMatches to remove
+            
+            if feature in exactMatches:
+                cleanTable = cleanTable.drop([feature], axis=1)
+                # print('{feature} due to being an exact match'.format(feature=feature))
+                continue
+            
+            # Check if the values in a column are all the same, if so we remove
+            
+            if isInvariant and cleanTable[feature].nunique() <= 1:
+                # print('{feature} due to being an invariant column.'.format(feature=feature))
+                cleanTable = cleanTable.drop([feature], axis=1)
+                continue
+                
+            # Else compare to regular expressions
+                
+            for regex in regexes:
+                if re.match(regex, feature):
+                    cleanTable = cleanTable.drop([feature], axis=1)
+                    # print('{feature} due to being a match with regular expression: {regex}'.format(feature=feature, regex=regex))
+                    break
+
+        if removeDuplicates:
+            # Finds empty columns to prevent them from being dropped
+            emptyCol = []
+            for column in cleanTable:
+                # Convert to numpy
+                data = cleanTable[column].to_numpy() 
+                if (data[0] == np.nan or data[0] == '') and (data[0] == data).all():
+                    emptyCol.append(column)
+                    
+            # Copy columns over to be added back after duplicates are removed
+            col = cleanTable[emptyCol]
+            
+            # Transpose the cleaned table and drop duplicate rows. Re-transpose to get back to the original table
+            cleanTable = cleanTable.T.drop_duplicates().T
+            
+            # Add empty columns back into table and reorder
+            cleanTable = pd.concat([cleanTable, col], axis=1)
+
+        return cleanTable
+
+    def binarizeData(self, table: pd.DataFrame, removeNaN: bool = True):
+        """[Function for binarizing a table given the presence or absence of data. i.e. if the first two columns of a dataframe row are empty followed by two present the binarized form would be 0011]
+
+        Args:
+            table (pd.DataFrame): [Pandas dataframe to binarize]
+            removeNaN (bool, optional): [Boolean that describes whether NaN values should be removed]. Defaults to True.
+
+        Returns:
+            [pd.Dataframe]: [Returns binarized table]
+        """
+
+        # Replace empty cells with NaN 
+        if removeNaN:
+            binTable = table.replace(r'^\s*$', np.nan, regex=True)
+
+        # Replace NaN values with 0 and all others with 1
+        binTable = binTable.notnull().astype('int')
+
+        return binTable
+
+    def getPresentColumns(self, signature: str, columns: List[str]):
```

Comment:
Python PEP (something) has naming standard 
- snake_casing for vars and method/func names
- PascalNaming (could be CamelCasing) for classes.
- UPPER_CASE for constants.